tools/daemonUtil.js

```diff
@@ -0,0 +1,94 @@
+const forever = require('forever-monitor');
+const waitOn = require('wait-on');
+const chalk = require('chalk');
+const samplesConfig = require('../.samples.config.json').oktaSample;
+
+const daemonUtil = module.exports;
+
+function startNpmScript(script, color) {
+  var child = new (forever.Monitor)(script, {
+    command: 'npm run',
+    max: 1,
+    silent: true
+  });
+
+  const normal = chalk[color || 'yellow'];
+  const bold = normal.bold.underline;
+
+  child.on('stdout', (msg) => {
+    console.log(normal(msg));
+  });
+
+  child.on('stderr', (msg) => {
+    console.log(bold(`Failed to start "npm run ${script}"`));
+    console.log(bold(msg));
+  });
+
+  child.on('exit', () => {
+    console.log(normal(`Finished "npm run ${script}"`));
+  });
+
+  return new Promise((resolve, reject) => {
+    child.on('start', () => {
+      resolve(child);
+    });
+    console.log(normal(`Running "npm run ${script}"`));
+    child.start();
+  });
+}
+
+function waitOnPromise(opts, context) {
```

Comment:
Nit - seems like the below code is all doing the same thing - could probably make this a "runAndWait" type function that just removes some of the boilerplate below.